pkg/ha/state/state.go

```diff
@@ -0,0 +1,77 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type State struct {
+	leader            string
+	leaseStart        time.Time
+	leaseUntil        time.Time
+	maxTimeSeen       time.Time // max data time seen by any instance
+	maxTimeInstance   string    // the instance name that’s seen the maxtime
+	maxTimeSeenLeader time.Time
+	_mu               sync.RWMutex
+}
+
+type StateView struct {
+	Leader            string
+	LeaseStart        time.Time
+	LeaseUntil        time.Time
+	MaxTimeSeen       time.Time
+	MaxTimeInstance   string
+	MaxTimeSeenLeader time.Time
+}
+
+// haLockState represents the current lock holder
+// as reported from the db
+type HALockState struct {
+	Cluster    string
+	Leader     string
+	LeaseStart time.Time
+	LeaseUntil time.Time
+}
+
+func (h *State) UpdateStateFromDB(latestState *HALockState, maxT time.Time, replicaName string) {
+	fmt.Println("latest state: ", latestState)
+	h._mu.Lock()
+	defer h._mu.Unlock()
+	h.leader = latestState.Leader
+	h.leaseStart = latestState.LeaseStart
+	h.leaseUntil = latestState.LeaseUntil
+	h.updateMaxTimeOnZero(maxT, replicaName)
```

Comment:
instead of calling this here why don't we set the initial maxTimeSeen&maxTimeInstance in a  NewState function that sets this upon creation of the state. And call this from ha_service.go:78